spec/client_spec.rb

```diff
@@ -39,5 +39,24 @@
         client.address.create(Fixture.ca_address1)
       }.to raise_error(StandardError)
     end
+
+    it 'calls custom client exec' do
+      client = described_class.new(
+        api_key: ENV['EASYPOST_TEST_API_KEY'],
+        custom_client_exec: lambda { |method, uri, headers, open_timeout, read_timeout, body = nil|
+          expect(method).to eq(:get)
+          expect(uri).to be_a(URI)
+          expect(headers).to be_a(Hash)
+          expect(open_timeout).to eq(30)
+          expect(read_timeout).to eq(60)
+          expect(body).to be_nil
+          return OpenStruct.new(status_code: 401, body: '{}')
+        },
+      )
+
+      expect {
+        client.address.retrieve('adr_123')
+      }.to raise_error(StandardError) # should throw error because our lambda returns 401
```

Comment:
question: can we test a more explicit error here? One of our own newly added ones?